db-server/test/backend/db_tests.js

```diff
@@ -54,1938 +54,1313 @@ function createEmail(data) {
-  tokenId : ACCOUNT_RESET_TOKEN_ID,
-  data : hex32(),
-  uid : ACCOUNT.uid,
-  createdAt: now + 5
+function makeMockChangePasswordToken(uid) {
+  const token = {
+    data: hex32(),
+    uid: uid,
+    createdAt: Date.now()
+  }
+  token.tokenId = hex32()
+  return token
 }
 
-function makeMockSessionToken(uid) {
-  var sessionToken = {
-    data : hex32(),
-    uid : uid,
-    createdAt : now + 1,
-    uaBrowser : 'mock browser',
-    uaBrowserVersion : 'mock browser version',
-    uaOS : 'mock OS',
-    uaOSVersion : 'mock OS version',
-    uaDeviceType : 'mock device type',
-    mustVerify: true,
-    tokenVerificationId : hex16(),
-    tokenVerificationCode: unblockCode(),
-    tokenVerificationCodeExpiresAt: Date.now() + 20000
+function makeMockAccountResetToken(uid, tokenId) {
+  const token = {
+    tokenId: tokenId || hex32(),
+    data: hex32(),
+    uid: uid,
+    createdAt: now + 5
   }
-
-  return sessionToken
+  return token
 }
 
 // To run these tests from a new backend, pass the config and an already created
 // DB API for them to be run against.
-module.exports = function(config, DB) {
+module.exports = function (config, DB) {
   describe('db_tests', () => {
 
-    let db
+    let db, accountData
     before(() => {
-      return DB.connect(config).then(db_ => {
-        db = db_
-        return db.ping()
-      })
-    })
-
-    it(
-      'account creation and password checking',
-      () => {
-        var emailBuffer = Buffer(ACCOUNT.email)
-        return db.accountExists(emailBuffer)
-        .then(function(exists) {
-          assert(false, 'account should not yet exist for this email address')
-        }, function(err) {
-          // ok, account could not be found'
+      return DB.connect(config)
+        .then(db_ => {
+          db = db_
+          return db.ping()
         })
-        .then(function() {
-          return db.createAccount(ACCOUNT.uid, ACCOUNT)
-        })
-        .then(function(account) {
-          assert.deepEqual(account, {}, 'Returned an empty object on account creation')
-          var emailBuffer = Buffer(ACCOUNT.email)
-          return db.accountExists(emailBuffer)
-        })
-        .then(function(exists) {
-          assert(exists, 'account exists for this email address')
+        .then(() => {
+          accountData = createAccount()
+          return db.createAccount(accountData.uid, accountData)
         })
-        .then(function() {
-          return db.account(ACCOUNT.uid)
+    })
+
+    describe('account', () => {
+      let accountData
+      beforeEach(() => {
+        accountData = createAccount()
+        return db.accountExists(accountData.emailBuffer)
+          .then(() => assert.fail, (err) => assert.equal(err.code, 404, 'Not found'))
+          .then(() => db.createAccount(accountData.uid, accountData))
+      })
+
+      describe('db.createAccount', () => {
+        it('should create account', () => {
+          accountData = createAccount()
```

Comment:
This shouldn't be here, right? The same code is already in the parent `beforeEach`.